### PR TITLE
Needed stuff for skeleton (spotless, editorconfig, recommended extensions, github actions)

### PR DESCRIPTION
**Pull request by obj-obj**

### CODE REVIEW

**Comment by obj-obj** [comment]
Someone review it pls so I dont have to deal with merge conflicts

**Comment by KangarooKoala** [comment]
These changes do need to be done ASAP, but we should also discuss these formatting changes with the whole programming team.
Formatting changes are tab-indent, enforcing import order (instead of just removing imports), and switching to Google java format (instead of Eclipse), right? Also, the java formatting no longer excludes `**/build/**`, `**/build-*/**`, `**/bin/**`, and `/.*`, though that may not actually have an effect. Groovy gradle also no longer has spotless formatting specified.

**Comment by obj-obj** [comment]
> These changes do need to be done ASAP, but we should also discuss these formatting changes with the whole programming team. Formatting changes are tab-indent, enforcing import order (instead of just removing imports), and switching to Google java format (instead of Eclipse), right? Also, the java formatting no longer excludes `**/build/**`, `**/build-*/**`, `**/bin/**`, and `/.*`, though that may not actually have an effect. Groovy gradle also no longer has spotless formatting specified.

Yeah there's no java files in those directories (only JSON) so I left it out
As for gradle, I removed it because it was using the painful eclipse formatter. We only have 1 gradle file that we barely change, so imo we don't really need formatting for that one file anyway.

**Comment by KangarooKoala** [comment]
I'm not saying this shouldn't be done, just that it (mostly the tab formatting) needs to be discussed with all of the programmers. (Also there's some spots where 2 space indents got deleted)

**Comment by KangarooKoala** [comment]
Also, why wasn't the automated test step included from the [2022 check](https://github.com/robototes/RapidReact2022/blob/main/.github/workflows/gradle.yml)?

**Comment by obj-obj** [comment]
> (Also there's some spots where 2 space indents got deleted)

I think that might be the formatter...

> Also, why wasn't the automated test step included from the [2022 check](https://github.com/robototes/RapidReact2022/blob/main/.github/workflows/gradle.yml)?

Oh I forgot about our previous github action and tried to make my own lmao, I'll just copy paste from last year like a true programmer now

**Comment by obj-obj** [comment]
Also leaving the gradle daemon on was intentional here - if you're running more than 1 gradle command in the same build script it's more efficient to use the daemon

**Comment by obj-obj** [comment]
astro

**Comment by obj-obj** [comment]
astro

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+	"recommendations": [
```
Comment (jbko6) [inline .vscode/extensions.json:2]: I don't think this is necessary for our project at our scale. I'd rather leave the extensions up to everyone else's preferences.

Comment (obj-obj) [reply .vscode/extensions.json:2]: It doesn't force you to install them, it just gives a popup that people new to vscode can use to easily install all the extensions

Comment (jbko6) [reply .vscode/extensions.json:2]: I'm aware, but those pop-ups are annoying tbh. Also, some of these extensions won't even be able to be used by the common programmer on their school laptop, such as GitLens. I don't want to confuse them

**Review by jbko6 — APPROVED** [review]

**Pull request merged**